src/helpers/makeA11yQuery.ts

```diff
@@ -31,8 +65,27 @@ const makeA11yQuery = <P extends unknown, M extends unknown>(
   queryNames: QueryNames,
   matcherFn: (prop: P, value: M) => boolean
 ) => (instance: ReactTestInstance) => {
-  const getBy = (matcher: M) => {
+  const filterWithName = (
+    node: ReactTestInstance,
+    options: QueryOptions,
+    matcher: M
+  ) => {
+    const matchesRole =
+      isNodeValid(node) && matcherFn(node.props[name], matcher);
+
+    return (
+      matchesRole && !!getQueriesForElement(node).queryByText(options.name)
```

Comment:
it looks like we should match for either `queryByText(options.name)` or `queryByLabelText(options.name)`